### PR TITLE
fix sink crashloop when upgrading from 0.1 to 0.2 

### LINKED ISSUE #366: Upgrade to v0.2.0 requires recreating event listeners
# Expected Behavior
Upgrading to triggers v0.2.0. Event listeners should be up and running.

# Actual Behavior
Event listeners are in crashloop, with the following error:
```
2020/01/20 14:18:55 Error loading logging configuration: lstat /etc/config-logging: no such file or directory

```

# Steps to Reproduce the Problem

1. Run triggers v0.1.0
2. Create event listeners
3. Upgrade to triggers v0.2.0

# Additional Info
To fix this issue:
- [ ] Update Release notes to say that the Params field has been removed and explicitly mention that Params field will not work at all and users will have to create `TriggerBindings` with static values instead.
- [ ]  Add a `DeperecatedParams` field to fix the sink `CrashLoopBackoff`

### PR DESCRIPTION
**Pull request by dibyom**
# Changes

Fixes #366 

The crashloop is due to 2 reasons:
1. We deprecated and removed the `params` field from the EventListener. So, when the reconciler tries to  unmarshal an old EventListener, it fails since it cannot parse the deleted `params` field.

2. The new event listener sink image expects `/etc/config-logging` to be mounted. However the reconciler code does not add the `volumeMount` for any existing sink deployments. So, the new pods go into a crashloop

To fix this, this commit does the following:
1. Add back the `params` field. Also, update the mutating webhook to delete any set `params` field in existing eventlisteners.
2. Add volumeMount for existing deployments if not present.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/triggers/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```
fixes a bug where upgrading an existing event listener from Triggers 0.1 to 0.2 causes a crashloop
```

### CODE REVIEW

**Comment by ncskier** [comment]
I just tested your branch. I can verify that I no longer see the EventListener running into a CrashLoop when I upgrade from v0.1.0 to this branch.

**Comment by ncskier** [comment]
When I updated my Triggers version, my EventListener's fields did not update (`binding` -> `bindings` and `params` getting deleted). Also, in the [EventListener reconciler file](https://github.com/tektoncd/triggers/blob/master/pkg/reconciler/v1alpha1/eventlistener/eventlistener.go), I don't see `Update()` called on the EventListener, so where do we update it with the changes from `event_listener_defaults.go`?

File: pkg/reconciler/v1alpha1/eventlistener/eventlistener.go [hunk 1]
```diff
@@ -354,6 +354,10 @@ func (c *Reconciler) reconcileDeployment(el *v1alpha1.EventListener) error {
 				existingDeployment.Spec.Template.Spec.Containers[0].Command = nil
 				updated = true
 			}
+			if len(existingDeployment.Spec.Template.Spec.Containers[0].VolumeMounts) == 0 {
+				existingDeployment.Spec.Template.Spec.Containers[0].VolumeMounts = container.VolumeMounts
+				updated = true
+			}
```
Comment (ncskier) [inline pkg/reconciler/v1alpha1/eventlistener/eventlistener.go]: Do we also need to check if the `existingDeployment.Spec.Template.Spec.Volumes` has been updated?

Comment (ncskier) [reply pkg/reconciler/v1alpha1/eventlistener/eventlistener.go]: I think we should add this below line 360:
```go
if !reflect.DeepEqual(existingDeployment.Spec.Template.Spec.Volumes, deployment.Spec.Template.Spec.Volumes) {
	existingDeployment.Spec.Template.Spec.Volumes =  deployment.Spec.Template.Spec.Volumes
	updated = true
}
```

Comment (dibyom) [reply pkg/reconciler/v1alpha1/eventlistener/eventlistener.go]: Added VolumeMount but not the actual Volume so the deployment is invalid :face_with_head_bandage:

**Comment by dibyom** [comment]
Ok, so the crashloop should be fixed though we still need to address @ncskier 's comment https://github.com/tektoncd/triggers/pull/369#issuecomment-577347870

**Comment by dibyom** [comment]
quick and dirty script I used to test this: https://gist.github.com/dibyom/93b8455ee18e496d408ec4ef51d2b476

**Comment by ncskier** [comment]
@googlebot I consent.

**Comment by ncskier** [comment]
@googlebot I consent

File: cmd/webhook/main.go [hunk 2]
```diff
@@ -82,7 +82,7 @@ func main() {
 
 	// Decorate contexts with the current state of the config.
 	ctxFunc := func(ctx context.Context) context.Context {
-		return ctx
+		return v1alpha1.WithUpgradeViaDefaulting(ctx)
 	}
```
Comment (ncskier) [inline cmd/webhook/main.go]: @dibyom I think `golangci` is complaining about this. It should be
```go
ctxFunc := v1alpha1.WithUpgradeViaDefaulting
```

**Pull request merged**